### PR TITLE
fix: sfu migration issues

### PR DESCRIPTION
**Pull request by raviteja83**
### Details(context, link the issue, how was the bug fixed, what does the new feature do)

- Duplicate tiles on migration
- Screenshare not getting removed
- add audio/video plugins on migration

### CODE REVIEW

File: packages/hms-video-store/src/media/tracks/HMSLocalVideoTrack.ts [hunk 1]
```diff
@@ -90,14 +90,25 @@ export class HMSLocalVideoTrack extends HMSVideoTrack {
...
       this.source!,
       this.eventBus,
       this.settings,
       this.room,
     );
+    if (this.pluginsManager.pluginsMap.size > 0) {
+      this.pluginsManager.pluginsMap.forEach(value => {
+        track
+          .addPlugin(value)
+          .catch((e: Error) => HMSLogger.e(this.TAG, 'Plugin add failed while migrating', value, e));
```
Comment (KaustubhKumar05) [inline packages/hms-video-store/src/media/tracks/HMSLocalVideoTrack.ts:105]: Can we add this to the audioTrack as well?

File: packages/hms-video-store/src/notification-manager/managers/TrackManager.ts [hunk 2]
```diff
@@ -205,7 +204,6 @@ export class TrackManager {
       track.type === HMSTrackType.AUDIO
         ? this.eventBus.audioTrackAdded.publish({ track: track as HMSRemoteAudioTrack, peer: hmsPeer as HMSRemotePeer })
         : this.listener?.onTrackUpdate(HMSTrackUpdate.TRACK_ADDED, track, hmsPeer);
-      this.store.removeTrackState(track.trackId);
```
Comment (KaustubhKumar05) [inline packages/hms-video-store/src/notification-manager/managers/TrackManager.ts:208]: Why was this removed?

File: packages/hms-video-store/src/transport/index.ts [hunk 3]
```diff
@@ -431,36 +431,27 @@ export default class HMSTransport {
...
-    this.createPeerConnections();
-    await this.negotiateOnFirstPublish();
+
     const localPeer = this.store.getLocalPeer();
     if (!localPeer) {
       return;
     }
-
-    let tracksToPublish = [];
+    console.log('creating peer connections');
```
Comment (KaustubhKumar05) [inline packages/hms-video-store/src/transport/index.ts:451]: Can be removed?

**Review by KaustubhKumar05 — APPROVED** [review]

File: packages/hms-video-store/src/transport/index.ts [hunk 4]
```diff
@@ -599,11 +589,9 @@ export default class HMSTransport {
...
     stream.removeSender(track);
     await p;
-    if (!sfuMigration) {
-      await track.cleanup();
-      // remove track from store on unpublish
-      this.store.removeTrack(track);
-    }
+    await track.cleanup();
+    // remove track from store on unpublish
+    this.store.removeTrack(track);
```
Comment (eswarclynn) [inline packages/hms-video-store/src/transport/index.ts:594]: Should we remove track state also on unpublish?

Comment (raviteja83) [reply packages/hms-video-store/src/transport/index.ts:594]: it is deleted above

File: packages/hms-video-store/src/notification-manager/managers/TrackManager.ts (hunk 2 above)
Comment (raviteja83) [reply packages/hms-video-store/src/notification-manager/managers/TrackManager.ts:208]: this should not be done from here. this trackState is needed when track is removed

**Review by eswarclynn — APPROVED** [review]

**Review by KaustubhKumar05 — APPROVED** [review]
